### PR TITLE
feat: Use secrets in macros

### PR DESCRIPTION
**Pull request by sbchaos**
Allows users to use secrets in assets and job configuration.
Exposes a different config `.secret` so that sensitive configuration are not printed in logs.

### CODE REVIEW

**Review by sravankorumilli — COMMENTED** [review]
Lets discuss over a call too, it is better to revisit this entirely, even after refactoring it is still hard to understand.

File: api/handler/v1beta1/runtime.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ func (sv *RuntimeServiceServer) RegisterInstance(ctx context.Context, req *pb.Re
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "%s: failed to register instance of jobrun %s", err.Error(), jobRun)
 	}
-	envMap, fileMap, err := sv.runSvc.Compile(ctx, namespaceSpec, jobRun, instance)
+	assets, err := sv.runSvc.Compile(ctx, namespaceSpec, jobRun, instance)
```
[the remark this answers is not in the document]
Comment (sbchaos) [reply api/handler/v1beta1/runtime.go:175]: Sure, we can have discussion around these to simplify more.

File: mock/instance.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func (s *RunService) Register(ctx context.Context, namespace models.NamespaceSpe
 	return args.Get(0).(models.InstanceSpec), args.Error(1)
 }
 
-func (s *RunService) Compile(ctx context.Context, namespaceSpec models.NamespaceSpec, jobRun models.JobRun, instanceSpec models.InstanceSpec) (envMap map[string]string, fileMap map[string]string, err error) {
+func (s *RunService) Compile(ctx context.Context, namespaceSpec models.NamespaceSpec, jobRun models.JobRun, instanceSpec models.InstanceSpec) (assets *models.CompiledAssets, err error) {
```
[the remark this answers is not in the document]
Comment (sbchaos) [reply mock/instance.go:127]: Sure, will rename to JobRunInput

File: models/instance.go [hunk 3]
```diff
@@ -22,7 +22,8 @@ const (
 	InstanceDataTypeFile = "file"
 
 	// InstanceDataTypeEnvFileName is run data env type file name
-	InstanceDataTypeEnvFileName = ".env"
+	InstanceDataTypeEnvFileName    = ".env"
```
[the remark this answers is not in the document]
Comment (sbchaos) [reply models/instance.go:25]: Sure, `.config` sounds better

[the remark this answers is not in the document]
Comment (sbchaos) [reply models/instance.go:25]: Will make this change in later pr

File: run/context.go [hunk 4]
```diff
@@ -156,105 +166,70 @@ func (fm *ContextManager) compileTemplates(templateValueMap, templateContext map
...
 
-func (fm *ContextManager) getProjectConfigMap() map[string]string {
-	configMap := map[string]string{}
-	for key, val := range fm.namespace.ProjectSpec.Config {
-		configMap[key] = val
-	}
-	return configMap
-}
+func (fm *ContextManager) createSecretsMap() map[string]string {
+	secretsMap := map[string]string{}
```
[the remark this answers is not in the document]
Comment (sbchaos) [reply run/context.go:188]: Changed

**Review by sravankorumilli — APPROVED** [review]

**Pull request merged**